vic/drivers/image/src/vic_init.c

```diff
@@ -1101,15 +1109,253 @@ vic_init(void)
+                }
+                if (lake_con[i].Cl[0] > 0) {
+                    log_err("cell %zu lake_idx is %d (lake not present) "
+                            "which requires max area fraction to be 0, but "
+                            "max area fraction is %f\n", i,
+                            lake_con[i].lake_idx, lake_con[i].Cl[0]);
+                }
+            }
+            else {
+                if (!(lake_con[i].z[0] > 0)) {
+                    log_err("cell %zu lake basin max depth is %f but must "
+                            "be > 0.\n", i, lake_con[i].z[0]);
+                }
+                else if (!(lake_con[i].mindepth <= lake_con[i].z[0])) {
+                    log_err("cell %zu lake basin mindepth is %f but "
+                            "must be <= max depth of %f\n",
+                            i, lake_con[i].mindepth, lake_con[i].z[0]);
+                }
+                if (!(lake_con[i].Cl[0] > 0 && lake_con[i].Cl[0] <= 1)) {
+                    log_err("cell %zu lake basin max area fraction is %f but "
+                            "we must have 0 < max area fraction < 1\n", i,
+                            lake_con[i].Cl[0]);
+                }
+                if (fabs(1 - lake_con[i].Cl[0] /
+                         veg_con[i][lake_con[i].lake_idx].Cv) > 0.01) {
+                    log_err("cell %zu lake basin max area fraction is %f but "
+                            "must == area fraction of veg tile containing "
+                            "lake (%f)\n", i, lake_con[i].Cl[0],
+                            veg_con[i][lake_con[i].lake_idx].Cv);
+                }
+                else {
+                    lake_con[i].Cl[0] = veg_con[i][lake_con[i].lake_idx].Cv;
+                }
+            }
+
+            // valdate other nodes
+            if (options.LAKE_PROFILE) {
+                for (j = 1; j < lake_con[i].numnod; j++) {
+                    if (!(lake_con[i].z[j] > 0 &&
+                          lake_con[i].z[j] < lake_con[i].z[j - 1])) {
+                        log_err("cell %zu lake basin node %zu depth is %f "
+                                "but must be > 0 and < node %zu depth %f\n",
+                                i, j, lake_con[i].z[j], j - 1,
+                                lake_con[i].z[j - 1]);
+                    }
+                    if (!(lake_con[i].Cl[j] > 0 &&
+                          lake_con[i].Cl[j] < lake_con[i].Cl[j - 1])) {
+                        log_err("cell %zu lake basin node %zu area fraction "
+                                "is %f but must be > 0 and < node %zu area "
+                                "fraction %f\n", i, j, lake_con[i].Cl[j], j - 1,
+                                lake_con[i].Cl[j - 1]);
+                    }
+                }
+            }
+        }
+
+        // compute other lake parameters here
+        for (i = 0; i < local_domain.ncells_active; i++) {
+            compute_cell_area(&(soil_con[i]));
```

Comment:
Question: is it sufficient for now to just set soil_con[i].cell_area = global_domain.locations[i].area?

The classic driver doesn't as yet have a global_domain structure, right?  If it did, we could get rid of soil_con.cell_area across the board.  Either way, it seems like this would be a separate isue...